tests/common/runner.rs

```diff
@@ -50,30 +53,100 @@ pub fn test_execute_revm<S: Storage + Clone + Send + Sync>(
 ) {
     let concurrency_level = thread::available_parallelism().unwrap_or(NonZeroUsize::MIN);
     assert_execution_result(
-        pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
-        pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
+        &pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
+        &pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
         false, // TODO: Parameterize this
     );
 }
 
+fn calculate_receipt_root(receipt_envelopes: Vec<ReceiptEnvelope>) -> B256 {
+    // create a BTreeMap (order must be ascending)
+    let entries = receipt_envelopes
+        .into_iter()
+        .enumerate()
+        .map(|(index, receipt)| {
+            let key_buffer = alloy_rlp::encode_fixed_size(&index);
+            let mut value_buffer = Vec::new();
+            receipt.encode_2718(&mut value_buffer);
+            (key_buffer, value_buffer)
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    // calculate the MPT root
+    let mut hash_builder = alloy_trie::HashBuilder::default();
+    for (k, v) in entries {
+        hash_builder.add_leaf(alloy_trie::Nibbles::unpack(&k), &v);
+    }
+    hash_builder.root()
+}
+
+fn get_receipt_envelopes(
+    block: &Block,
+    tx_results: &[PevmTxExecutionResult],
+) -> Vec<ReceiptEnvelope> {
+    // get the receipts
+    let receipt_iter = tx_results.iter().map(|tx| tx.receipt.clone());
+
+    // get the tx_type list
+    let tx_type_iter = block
+        .transactions
+        .txns()
+        .unwrap()
+        .map(|tx| TxType::try_from(tx.transaction_type.unwrap_or_default()).unwrap());
+
+    // zip them to get the receipt envelopes
+    let receipt_envelope_iter =
+        Iterator::zip(receipt_iter, tx_type_iter).map(|(receipt, tx_type)| match tx_type {
+            TxType::Legacy => ReceiptEnvelope::Legacy(receipt.with_bloom()),
+            TxType::Eip2930 => ReceiptEnvelope::Eip2930(receipt.with_bloom()),
+            TxType::Eip1559 => ReceiptEnvelope::Eip1559(receipt.with_bloom()),
+            TxType::Eip4844 => ReceiptEnvelope::Eip4844(receipt.with_bloom()),
+        });
+    receipt_envelope_iter.collect::<Vec<_>>()
+}
+
 // Execute an Alloy block sequentially & with PEVM and assert that
 // the execution results match.
 pub fn test_execute_alloy<S: Storage + Clone + Send + Sync>(
     storage: S,
     block: Block,
     parent_header: Option<Header>,
     must_succeed: bool,
+    must_check_receipts_root: bool,
```

Comment:
`must_match_receipts_root` sounds better. Also, we can now remove `must_succeed` from here and `assert_execution_result`. This supersedes `must_succeed` as it needs to `unwrap` an `Ok` out fo the execution results to calculate the receipts root anyway.

Reply:
`must_succeed` seems to be still needed, doesn't it?

see:
```
fn raw_transfers_independent_alloy()
fn empty_alloy_block()
fn one_tx_alloy_block()
```

Reply:
or do you mean that we should always treat it as `must_succeed = true` and hence `must_succeed` is not needed?

Reply:
We can remove `must_succeed` now because it only served mainnet blocks that are now served by `must_match_receipts_root`. The mocked Alloy blocks inheriting `must_succeed = true` is only a small artifact and their `test_execute_revm`'s counterpart has `must_succeed` hardcoded to `false` anyway.

If one day we want `must_succeed` for mocked blocks then we'll redo it properly for both mocked REVM blocks & Alloy blocks. For now the extra complexity isn't worth it.